File: TransCoder_tokenized_test_set_functions/cpp/QUEUE_BASED_APPROACH_FOR_FIRST_NON_REPEATING_CHARACTER_IN_A_STREAM.cpp
void firstnonrepeating ( char str [ ] ) {
  queue < char > q;
  int charCount [ MAX_CHAR ] = {
    0 };
    for ( int i = 0;
    str [ i ];
    i ++ ) {
      q . push ( str [ i ] );
      charCount [ str [ i ] - 'a' ] ++;
      while ( ! q . empty ( ) ) {
        if ( charCount [ q . front ( ) - 'a' ] > 1 ) q . pop ( );
        else {
          cout << q . front ( ) << " ";
          break;
        }
      }
      if ( q . empty ( ) ) cout << - 1 << " ";
    }
    cout << endl;
  }

File: TransCoder_tokenized_test_set_functions/java/QUEUE_BASED_APPROACH_FOR_FIRST_NON_REPEATING_CHARACTER_IN_A_STREAM.java
static void firstNonRepeating ( String str ) {
  int [ ] charCount = new int [ MAX_CHAR ];
  Queue < Character > q = new LinkedList < Character > ( );
  for ( int i = 0;
  i < str . length ( );
  i ++ ) {
    char ch = str . charAt ( i );
    q . add ( ch );
    charCount [ ch - 'a' ] ++;
    while ( ! q . isEmpty ( ) ) {
      if ( charCount [ q . peek ( ) - 'a' ] > 1 ) q . remove ( );
      else {
        System . out . print ( q . peek ( ) + " " );
        break;
      }
    }
    if ( q . isEmpty ( ) ) System . out . print ( - 1 + " " );
  }
  System . out . println ( );
}